geogfn: fix segmentize of >MaxInt64 coordinates

otan opened the pull request:
Release note (bug fix): Error gracefully when attempting to run
ST_Segmentize and generating more than MaxInt64 points on a GEOGRAPHY.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/63263)
<!-- Reviewable:end -->

otan committed:
geogfn: fix segmentize of >MaxInt64 coordinates

Release note (bug fix): Error gracefully when attempting to run
ST_Segmentize and generating more than MaxInt64 points on a GEOGRAPHY.

otan commented:
bors r=andyyang890
